processor/src/trace/virtual_bus/sum_check/domain.rs

```diff
@@ -0,0 +1,65 @@
+use alloc::vec::Vec;
+use vm_core::FieldElement;
+use winter_prover::math::batch_inversion;
+
+/// Implements barycentric evaluation where the interpolation domain is `0, 1, ..., max_degree`
+/// where `max_degree` is maximal polynomial supported degree.
+pub struct EvaluationDomain<E>
+where
+    E: FieldElement,
+{
+    interpolation_points: Vec<E>,
+    barycentric_weights: Vec<E>,
+}
+
+impl<E> EvaluationDomain<E>
+where
+    E: FieldElement,
+{
+    pub fn new(max_degree: usize) -> Self {
+        let interpolation_points: Vec<E> = (0..=max_degree).map(|x| E::from(x as u32)).collect();
+        let barycentric_weights = barycentric_weights(&interpolation_points);
+
+        Self {
+            interpolation_points,
+            barycentric_weights,
+        }
+    }
+
+    pub fn evaluate(&self, evaluations: &[E], r: E) -> E {
+        evaluate_barycentric(&self.interpolation_points, evaluations, r, &self.barycentric_weights)
+    }
+}
+
+/// Computes the barycentric weights for a set of interpolation points.
+pub fn barycentric_weights<E: FieldElement>(points: &[E]) -> Vec<E> {
+    let n = points.len();
+    let tmp = (0..n)
```

Comment:
nit: would rename to `weight_denominators` or something of the like

Reply:
changed

Reply:
Actually I was referring to the `tmp` variable to be renamed to `weight_denominators`, since it contains the denominators before they get inverted. I think `barycentric_weights()` is a good name for the function, since it agrees with the definition of barycentric weights [in the literature](https://people.maths.ox.ac.uk/trefethen/barycentric.pdf).